### PR TITLE
Library filter UI change

### PR DESCRIPTION
**Pull request by CarlosEsco**
#1157 

@LinkCable  styled it same as Catalog is expandable/collapsible necessary?  If so will have to roll our own custom solution for this area. 
image

### CODE REVIEW

**Comment by LinkCable** [comment]
@CarlosEsco I would say it would be preferable, both so it would match the Categories filter list and so that we could incorporate stuff like https://github.com/inorichi/tachiyomi/issues/1147 without lots of scrolling down.

If it's a hassle though I don't wanna put more work on you.

**Comment by CarlosEsco** [comment]
I think for now just having similar visuals will help.  Adding  expand/collapse can be done later

**Comment by inorichi** [comment]
Please fix the commented code, unless they are there for a reason, but I believe they were left there while testing.

**Comment by CarlosEsco** [comment]
@inorichi removed them.  I was waiting on LinkCables thoughts of expandable/collapsible.  Also should i remove DrawerSwipeCloseListener?  Its not used anymore anywhere.

**Comment by LinkCable** [comment]
@CarlosEsco I'd like it, but like you said we can add it later.

**Comment by Flat** [comment]
I think for consistency this filter menu should also be openable by sliding it out from the right, the same way the menu works for catalogues.

**Comment by CarlosEsco** [comment]
It does now

**Comment by Flat** [comment]
Ah I missed that. Cheers

**Comment by LinkCable** [comment]
> It does now

Does it interfere with swiping across categories at all in your testing?

**Comment by CarlosEsco** [comment]
Just checked and no like the catalog swipe the start has to be right on the edge to open drawer.   The category swipe can be anywhere else

**Comment by LinkCable** [comment]
@inorichi @NoodleMage code review/merge?

File: app/src/main/java/eu/kanade/tachiyomi/ui/library/LibraryController.kt [hunk 1]
```diff
@@ -177,11 +176,8 @@ class LibraryController(
 
     override fun createSecondaryDrawer(drawer: DrawerLayout): ViewGroup {
         val view = drawer.inflate(R.layout.library_drawer) as LibraryNavigationView
-        drawerListener = DrawerSwipeCloseListener(drawer, view).also {
-            drawer.addDrawerListener(it)
-        }
         navView = view
-        drawer.setDrawerLockMode(DrawerLayout.LOCK_MODE_LOCKED_CLOSED, Gravity.END)
+        drawer.setDrawerLockMode(DrawerLayout.LOCK_MODE_UNLOCKED, Gravity.END)
```
Comment (NoodleMage) [inline app/src/main/java/eu/kanade/tachiyomi/ui/library/LibraryController.kt]: Keep this LOCK_MODE_LOCKED_CLOSED. 
LOCK_MODE_UNLOCKED gives can give weird behavior when switching categories.

File: app/src/main/java/eu/kanade/tachiyomi/widget/ExtendedNavigationView.kt [hunk 2]
```diff
@@ -193,7 +192,7 @@ open class ExtendedNavigationView @JvmOverloads constructor(
         @CallSuper
         override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): Holder {
             return when (viewType) {
-                VIEW_TYPE_HEADER -> HeaderHolder(parent)
+                VIEW_TYPE_HEADER -> HeaderHolder(parent, onClick)
```
Comment (NoodleMage) [inline app/src/main/java/eu/kanade/tachiyomi/widget/ExtendedNavigationView.kt]: What's the reason for onClick?.

File: app/src/main/java/eu/kanade/tachiyomi/ui/library/LibraryController.kt (hunk 1 above)
Comment (CarlosEsco) [reply app/src/main/java/eu/kanade/tachiyomi/ui/library/LibraryController.kt]: So locking it you lose the consistency between this and the filter drawer in catalog.   I verified this and it only slides the drawer out when your right near the edge of the screen.

File: app/src/main/java/eu/kanade/tachiyomi/widget/ExtendedNavigationView.kt (hunk 2 above)
Comment (CarlosEsco) [reply app/src/main/java/eu/kanade/tachiyomi/widget/ExtendedNavigationView.kt]: not needed anymore

File: app/src/main/java/eu/kanade/tachiyomi/ui/library/LibraryController.kt (hunk 1 above)
Comment (NoodleMage) [reply app/src/main/java/eu/kanade/tachiyomi/ui/library/LibraryController.kt]: I will merge with LOCK_MODE_UNLOCKED. If it turns out it bothers people we revert it back to LOCK_MODE_LOCK_UNLOCKED, good with you?

**Comment by NoodleMage** [comment]
```:app:compileStandardDebugKotline: /home/travis/build/inorichi/tachiyomi/app/src/main/java/eu/kanade/tachiyomi/widget/ExtendedNavigationView.kt: (209, 28): Unresolved reference: title```

You broke something :)

**Comment by CarlosEsco** [comment]
fixed

**Pull request merged**